Client/mods/deathmatch/logic/CClientGame.cpp

```diff
@@ -4759,64 +4759,78 @@ bool CClientGame::VehicleDamageHandler(CEntitySAInterface* pVehicleInterface, fl
 
 bool CClientGame::ObjectDamageHandler(CObjectSAInterface* pObjectInterface, float fLoss, CEntitySAInterface* pAttackerInterface)
 {
-    if (pObjectInterface)
-    {
-        CPools*                   pPools = g_pGame->GetPools();
-        SClientEntity<CObjectSA>* pObjectEntity = pPools->GetObject((DWORD*)pObjectInterface);
-        if (pObjectEntity)
-        {
-            CClientObject* pClientObject = reinterpret_cast<CClientObject*>(pObjectEntity->pClientEntity);
-            if (!pClientObject)
-            {
-                return true;
-            }
-            CLuaArguments Arguments;
-            Arguments.PushNumber(fLoss);
+    if (!pObjectInterface)
+        return false;
 
-            CClientEntity* pClientAttacker = pPools->GetClientEntity((DWORD*)pAttackerInterface);
-            if (pClientAttacker)
-                Arguments.PushElement(pClientAttacker);
-            else
-                Arguments.PushNil();
+    CPools*                   pPools = g_pGame->GetPools();
+    SClientEntity<CObjectSA>* pObjectEntity = pPools->GetObject((DWORD*)pObjectInterface);
+    if (!pObjectEntity)
+        return false;
 
-            return pClientObject->CallEvent("onClientObjectDamage", Arguments, true);
-        }
+    CClientObject* pClientObject = reinterpret_cast<CClientObject*>(pObjectEntity->pClientEntity);
+    if (!pClientObject)
+        return true;
+
+    CLuaArguments Arguments;
+    Arguments.PushNumber(fLoss);
+
+    CClientEntity* pClientAttacker = pPools->GetClientEntity((DWORD*)pAttackerInterface);
+    ElementID      attackerID = INVALID_ELEMENT_ID;
+
+    if (pClientAttacker)
+    {
+        attackerID = pClientAttacker->GetID();
+        Arguments.PushElement(pClientAttacker);
     }
-    return true;
+    else
+        Arguments.PushNil();
+
+    bool bResult = pClientObject->CallEvent("onClientObjectDamage", Arguments, true);
+
+    if (bResult && !pClientObject->IsLocalEntity())
+        SendObjectDamagePacket(pClientObject, fLoss, attackerID);
+
+    return bResult;
```

Comment:
Every player in stream triggers this event.
This causes event flooding and performance drops

Reply:
Do you have an idea how to improve it?